src/Extraload/Transformer/MapFieldsTransformer.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Extraload\Transformer;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+class MapFieldsTransformer implements TransformerInterface
+{
+    private $fields;
+
+    public function __construct(array $fields)
+    {
+        $this->fields = $fields;
+    }
+
+    public function transform($data)
+    {
+        $result = [];
+
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+
+        foreach ($data as $key => $val) {
+            if (array_key_exists($key, $this->fields)) {
```

Comment:
`$propertyAccessor->isWritable($result, "[{$this->fields[$key]}]")`.

Reply:
This one:

```php
foreach ($data as $key => $val) {
    if ($propertyAccessor->isWritable($result, "[{$this->fields[$key]}]")) {
        $propertyAccessor->setValue($result, "[{$this->fields[$key]}]", $val);
    }
}
```
Causes:
```
[07-May-2018 20:29:54 Europe/London] PHP Notice:  Undefined index: isbn in /home/standard/projects/Extraload/src/Extraload/Transformer/MapFieldsTransformer.php on line 23
[07-May-2018 20:29:54 Europe/London] PHP Fatal error:  Uncaught Symfony\Component\PropertyAccess\Exception\InvalidPropertyPathException: Could not parse property path "[]". Unexpected token "[" at position 0 in /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyPath.php:116
Stack trace:
#0 /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyAccessor.php(886): Symfony\Component\PropertyAccess\PropertyPath->__construct('[]')
#1 /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyAccessor.php(297): Symfony\Component\PropertyAccess\PropertyAccessor->getPropertyPath('[]')
#2 /home/standard/projects/Extraload/src/Extraload/Transformer/MapFieldsTransformer.php(23): Symfony\Component\PropertyAccess\PropertyAccessor->isWritable(Array, '[]')
#3 /home/standard/projects/Extraload/src/Extraload/Pipeline/DefaultPipeline.php(38): Extraload\Transformer\MapFieldsTransformer->transform(Array)
#4 /home/standard/projects/Extraload/examples/doctrine/04_default_doctrine_query_map_fields_transformer_dbal_loader.php(21): Ex in /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyPath.php on line 116
```
And the following one:
```php
foreach ($data as $key => $val) {
    if ($propertyAccessor->isWritable($result, $this->fields[$key])) {
        $propertyAccessor->setValue($result, "[{$this->fields[$key]}]", $val);
    }
}
```
Says:
```
[07-May-2018 20:32:04 Europe/London] PHP Notice:  Undefined index: isbn in /home/standard/projects/Extraload/src/Extraload/Transformer/MapFieldsTransformer.php on line 23
[07-May-2018 20:32:04 Europe/London] PHP Fatal error:  Uncaught Symfony\Component\PropertyAccess\Exception\InvalidArgumentException: The property path constructor needs a string or an instance of "Symfony\Component\PropertyAccess\PropertyPath". Got: "NULL" in /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyPath.php:80
Stack trace:
#0 /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyAccessor.php(886): Symfony\Component\PropertyAccess\PropertyPath->__construct(NULL)
#1 /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyAccessor.php(297): Symfony\Component\PropertyAccess\PropertyAccessor->getPropertyPath(NULL)
#2 /home/standard/projects/Extraload/src/Extraload/Transformer/MapFieldsTransformer.php(23): Symfony\Component\PropertyAccess\PropertyAccessor->isWritable(Array, NULL)
#3 /home/standard/projects/Extraload/src/Extraload/Pipeline/DefaultPipeline.php(38): Extraload\Transformer\MapFieldsTransformer->transform(Array)
#4 /home/standard/projects/Extraload/examples/doctrine/04_default_doctrine_quer in /home/standard/projects/Extraload/vendor/symfony/property-access/PropertyPath.php on line 80
```

Reply:
Mmm. Perhaps in this particular example (a basic mapper) the `PropertyAccess` is not necessary. Apart from this one, should we write a few other transformers? :)

Reply:
Let me debug, what php script are you running?

Reply:
I am running `examples/doctrine/04_default_doctrine_query_map_fields_transformer_dbal_loader.php` 
with `PHP 7.1.17`